dbt/macros/nearest_pin_neighbors.sql

```diff
@@ -46,27 +35,32 @@
                     from
                         (
                             select
-                                dp.pin10,
-                                dp.x_3435,
-                                dp.y_3435,
+                                sp.pin10,
+                                sp.x_3435,
+                                sp.y_3435,
                                 loc.year,
                                 loc.pin10 as neighbor_pin10,
                                 st_distance(
-                                    st_point(dp.x_3435, dp.y_3435), loc.point
+                                    st_point(sp.x_3435, sp.y_3435), loc.point
                                 ) as dist
-                            from distinct_pins as dp
+                            from source_pins as sp
                             inner join
                                 pin_locations as loc
                                 on st_contains(
-                                    st_buffer(
-                                        st_point(dp.x_3435, dp.y_3435), {{ radius_km }}
-                                    ),
-                                    loc.point
+                                    st_buffer(sp.point, {{ radius_km }}), loc.point
                                 )
-                                and dp.pin10 != loc.pin10
+                            -- This horrifying conditional is designed to trick the
+                            -- Athena query
+                            -- planner. For some reason, adding a true conditional to
+                            -- a query with a
+                            -- spatial join (like the one above) results in terrible
+                            -- performance,
+                            -- while doing a cross join then filtering the rows is
+                            -- much faster
+                            where abs(cast(loc.year as int) - cast(sp.year as int)) = 0
                         ) as dists
                 )
-            where row_num <= 4
+            where row_num <= {{ num_neighbors }} + 1
```

Comment:
I think this was actually incorrect and should have been set dynamically using `num_neighbors`, correct me if I'm wrong.